pandas/core/indexes/datetimes.py

```diff
@@ -2241,21 +2241,56 @@ def date_range(start=None, end=None, periods=None, freq='D', tz=None,
     -------
     rng : DatetimeIndex
 
+    Examples
+    --------
+
+    Given the ``start`` and ``end`` mandatory parameters,
+    with default `freq='D'` (Days)
+
+    >>> pd1 = pd.date_range(start='1/1/2018', end='1/10/2018')
+    >>> pd1
+    DatetimeIndex(['2018-01-01', '2018-01-02', '2018-01-03', '2018-01-04',
+                   '2018-01-05', '2018-01-06', '2018-01-07', '2018-01-08',
+                   '2018-01-09', '2018-01-10'],
+                  dtype='datetime64[ns]', freq='D')
+
+    Given the ``start`` and ``periods`` mandatory parameters,
+    periods would be the limit of the DatetimeIndex, with default `freq='D'`
+    (D=calendar day frequency)
+
+    >>> pd2 = pd.date_range(start='1/1/2018', periods=5)
+    >>> pd2
+    DatetimeIndex(['2018-01-01', '2018-01-02', '2018-01-03', '2018-01-04',
+                   '2018-01-05'],
+                  dtype='datetime64[ns]', freq='D')
+
+    Changed the `freq` (frequency) to `'M'` (month end frequency) and `tz` to `'UTC'`
+
+    >>> pd3 = pd.date_range(start='1/1/2018', periods=5, freq='M', tz='UTC')
+    >>> pd3
+    DatetimeIndex(['2018-01-31', '2018-02-28', '2018-03-31', '2018-04-30',
+                   '2018-05-31'],
+                  dtype='datetime64[ns, UTC]', freq='M')
+
+    closed set to `'left'`, this makes the month range with the starting month
+    (inclusive of dec to feb)
+
+    >>> pd4 = pd.date_range(start='31/12/2017', end='3/1/2018', freq='M', closed='left')
+    >>> pd4
+    DatetimeIndex(['2017-12-31', '2018-01-31', '2018-02-28'], dtype='datetime64[ns]', freq='M')
+
+    closed set to `'right'`, this makes the month range after the starting month
+    (inclusive of jan to feb)
+
+    >>> pd4 = pd.date_range(start='31/12/2017', end='3/1/2018', freq='M', closed='right')
+    >>> pd4
+    DatetimeIndex(['2018-01-31', '2018-02-28'], dtype='datetime64[ns]', freq='M')
+
     See Also
     --------
     pandas.period_range : Return a fixed frequency PeriodIndex.
     pandas.interval_range : Return a fixed frequency IntervalIndex.
-
-    Examples
-    --------
-    >>> pd.date_range('2018-10-03', periods=2) # doctest: +NORMALIZE_WHITESPACE
-    DatetimeIndex(['2018-10-03', '2018-10-04'], dtype='datetime64[ns]',
-                   freq='D')
-
-    >>> pd.date_range(start='2018-01-01', end='20180103')
-    ... # doctest: +NORMALIZE_WHITESPACE
-    DatetimeIndex(['2018-01-01', '2018-01-02', '2018-01-03'],
-                    dtype='datetime64[ns]', freq='D')
+    Numpy daterange: `numpy daterange<https://docs.scipy.org/doc/numpy/reference/arrays.datetime.html>`__.
```

Comment:
See also goes before the examples. The order is the one in the sections of the doc: https://python-sprints.github.io/pandas/guide/pandas_docstring.html

If you use `numpy.arange` the link will be automatically generated. The `pandas.` prefix is nor needed, better to remove it. After the numpy one, you need a space before the colon, otherwise sphinx gets confused when building the doc. After the colon, we place a description, as the link is automatically build.  This doc explains it better:
https://python-sprints.github.io/pandas/guide/pandas_docstring.html#section-5-see-also

Reply:
Hey @datapythonista , 

I can provide a link to `numpy.arange`, but that page does not have good examples of date range, therefore, I was thinking to add a link.

Reply:
Agree, and sounds good. But as @TomAugspurger said, `See Also` is the right place for related methods/functions/classes, the link should be in the extended summary (or in references).